internal/pkg/cli/deploy/workload.go

```diff
@@ -367,24 +388,71 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 	if err != nil {
 		return fmt.Errorf("login to image repository: %w", err)
 	}
+
+	var digestsMu sync.Mutex
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+	bufferIdx := 0
+	labeledBuffers := make([]*syncbuffer.LabeledSyncBuffer, len(buildArgsPerContainer))
```

Comment:
nit: i'd rather get rid of the index and just use `append` - not worried about performance as this is a one time thing and it's a little more readable as there's already a lot of local variables here